solidity/dashboard/src/sagas/liquidity-rewards.js

```diff
@@ -0,0 +1,79 @@
+import { takeLatest, fork, call, put } from "redux-saga/effects"
+import BigNumber from "bignumber.js"
+import { getContractsContext, logError } from "./utils"
+import {
+  fetchStakedBalance,
+  fetchWrappedTokenBalance,
+  fetchLPRewardsTotalSupply,
+  fetchRewardBalance,
+} from "../services/liquidity-rewards"
+import { gt } from "../utils/arithmetics.utils"
+import { LIQUIDITY_REWARD_PAIR } from "../constants/constants"
+
+function* fetchAllLiquidtyRewardsData(action) {
+  const { address } = action.payload
+
+  for (const [pairName, value] of Object.entries(LIQUIDITY_REWARD_PAIR)) {
+    yield fork(fetchLiquidityRewardsData, { name: pairName, ...value }, address)
+  }
+}
+
+function* fetchLiquidityRewardsData(liquidityRewardPair, address) {
+  const contracts = yield getContractsContext()
+
+  const LPRewardsContract = contracts[liquidityRewardPair.contractName]
+  try {
+    yield put({
+      type: `liquidity_rewards/${liquidityRewardPair.name}_fetch_data_start`,
+      payload: { liquidityRewardPair: liquidityRewardPair.name },
+    })
+    // Fetching balance of liquidity token for a given uniswap pair deposited in
+    // the `LPRewards` contract.
+    const lpBalance = yield call(fetchStakedBalance, address, LPRewardsContract)
+    // Fetching balance of liquidity token for a given uniswap pair.
+    const wrappedTokenBalance = yield call(
+      fetchWrappedTokenBalance,
+      address,
+      LPRewardsContract
+    )
+    let reward = 0
+    let shareOfPoolInPercent = 0
+    if (gt(lpBalance, 0)) {
+      // Fetching available reward balance from `LPRewards` contract.
+      reward = yield call(fetchRewardBalance, address, LPRewardsContract)
+      // Fetching total deposited liqidity tokens in the `LPRewards` contract.
+      const totalSupply = yield call(
+        fetchLPRewardsTotalSupply,
+        LPRewardsContract
+      )
+      // % of total pool in the `LPRewards` contract.
+      shareOfPoolInPercent = new BigNumber(lpBalance)
+        .div(totalSupply)
+        .multipliedBy(100)
+        .toFixed(2, BigNumber.ROUND_DOWN)
```

Comment:
I think this would look better if we remove the trailing 0's. So this values:

- 70.00%
- 70.20%
- 70.25%

should be displayed as:

- 70%
- 70.2%
- 70.25%

Reply:
Also it probably would be better to format it in the view instead of here.

Reply:
ce6e7ea

Reply:
After thinking the whole thing I do agree with adding trailing format and rounding to two decimal places in the view. Since it's possible that the % of the total pool may be less than 0.01 so should we display `<0.01 %`, so we should store the total percentage value in store but convert it to a dedicated format in a component.

Reply:
7547b7c